### DISCUSSION TITLE
How to stop Repeat Animation

### DISCUSSION THREAD

**Discussion by suryanadiminti99** [sceneview/sceneform-android#41] [Q&A]
Hi @ThomasGorisse I am using below code for playing animation . Animation will play repeatedly . I want to play it one time. How can i define it . 

`FilamentAsset filamentAsset = arrayListModelNodes.get(j).getRenderableInstance().getFilamentAsset();

                                                    if (filamentAsset.getAnimator().getAnimationCount() > 0) {
                                                        // Loop animations
                                                        for (int k = 0;k<filamentAsset.getAnimator().getAnimationCount();k++){

                                                            Log.v("Plugxr","Animations : "+filamentAsset.getAnimator().getAnimationName(k));

                                                           animators.add(new AnimationInstance(filamentAsset.getAnimator(), k, System.nanoTime(),filamentAsset.getAnimator().getAnimationName(k),"1","0"));
                                                        }

                                                    }`

`// For animations
            Long time = System.nanoTime();

            for (AnimationInstance animator : animators) {
                // Start animation after delay time.
                final Handler handler = new Handler(Looper.getMainLooper());
                handler.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        //Do something after 100ms
                        animator.animator.applyAnimation(
                                animator.index,
                                (float) ((time - animator.startTime) / (double) SECONDS.toNanos(1))
                                        % animator.duration);

                        Log.v("Plugxr", String.valueOf(animator.name));

                        animator.animator.updateBoneMatrices();

                    }
                }, Long.parseLong(animator.delay_time));

            }`

**Comment by ThomasGorisse** [comment]
Just use this to start the animation with `repeat=false`:
```java
ObjectAnimator modelAnimator = model.getRenderableInstance().animate(false);
modelAnimator.start();
```
you can use this to stop it:
```java
modelAnimator.cancel();
```
